presto-native-execution/presto_cpp/main/PrestoTask.h

```diff
@@ -161,9 +163,9 @@ struct PrestoTask {
     return updateStatusLocked();
   }
 
-  protocol::TaskInfo updateInfo() {
+  protocol::TaskInfo updateInfo(bool summarize = false) {
```

Comment:
Maybe drop the optional parameter and make sure summarize is always passed. All methods invoking this should take into account `sumarize` flag.